src/no/runsafe/dergons/Dergon.java

```diff
@@ -70,8 +168,12 @@ private void bO()
 
 			for (IPlayer player : players)
 			{
-				// Skip the player if we're vanished or in creative mode.
-				if (player.isVanished() || player.getGameMode() == GameMode.CREATIVE || isRidingPlayer(player.getName()))
+				// Skip the player if we're vanished, in creative mode, or in spectator mode.
```

Comment:
Since we're using this clause multiple times, and it's quite a chunky statement, I feel like it would be best to compile it down into a function, something like `isValidPickupTarget()`, and call that instead of checking vanish, death and game-mode status. Obviously leave `isRidingPlayer` into this clause alone, since it's not used in the other.